prior-work/fraunhofer/directory-td.jsonld

```diff
@@ -0,0 +1,98 @@
+{
+    "@context": "https://www.w3.org/2019/wot/td/v1",
+    "title": "LinkSmart Thing Directory",
+    "securityDefinitions": {
+      "bearer_sc": {
+        "scheme": "bearer"
+      },
+      "basic_sc": {
+        "scheme": "basic"
+      }
+    },
+    "security": "bearer_sc",
+    "actions": {
+      "td": {
+        "description": "Add, modify, or remove a Thing Description",
+        "uriVariables": {
+          "id": {
+            "title": "Thing Description ID",
+            "type": "string",
+            "format": "iri-reference"
+          }
+        },
+        "forms": [
+          {
+            "href": "/td",
+            "htv:methodName": "POST",
+            "contentType": "application/ld+json"
+          },
+          {
+            "href": "/td/{id}",
+            "htv:methodName": "PUT",
+            "contentType": "application/ld+json"
+          },
+          {
+            "href": "/td/{id}",
+            "htv:methodName": "PATCH",
+            "contentType": "application/ld+json"
+          },
+          {
+            "href": "/td/{id}",
+            "htv:methodName": "DELETE"
+          }
+        ]
+      }
+    },
+    "properties": {
+      "td": {
+        "description": "Retrieve a Thing Description",
+        "uriVariables": {
+          "id": {
+            "title": "Thing Description ID",
+            "type": "string",
+            "format": "iri-reference"
+          }
+        },
+        "forms": [
+          {
+            "href": "/td/{id}",
+            "htv:methodName": "GET",
+            "contentType": "application/ld+json"
+          }
+        ]
+      },
```

Comment:
Using URL templates in the form of a property description is also kind of odd, because it's really a collection of multiple resources which is being described rather than a single property resource.

Reply:
IMO, TD has evolved for describing things and not collection services. Trying to use a TD to describe a collection API has its side effects. I decided to map "properties" to resources of the collection. I also thought about "getTD" and "listTD" as actions on the collection, but it seems contradictory to the specs because those operations don't manipulate the state of the collection.

"An Interaction Affordance that allows to invoke a function of the Thing, which **manipulates state**" [[1](https://www.w3.org/TR/wot-thing-description/#actionaffordance)]

Reply:
I agree that getTD and ListTD don't make sense as actions.